Document Controls, a closure and a FP from THM

GiudGiud opened the pull request:
refs #19737

This will make https://mooseframework.inl.gov/source/index.html#! look better. Less mixing of incompatible things

GiudGiud committed:
Make a dedicated folder for fluid properties in docs
They have their own syntax, it s only fair

GiudGiud committed:
Add documentation and description to linear-density fluid properties, refs #19737

GiudGiud committed:
Add object documentation for the simple 1 phase closure, refs #19737

GiudGiud committed:
Add object documentation for thm's control logic objects, refs #19737

GiudGiud committed:
Add class descriptions for THM's control logic

GiudGiud committed:
Move THM controls to their own control logic folders
They use a different syntax, they shouldnt be lumped with controls

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1568067/) on a9f048b wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/24589/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/24589/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 968eb46a3868f624d7930083ffdedd039b3d4fd5`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1568151/) on a019367 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24589/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1568192/) on a019367 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>968eb4</th>
      <th colspan=4>#24589 a01936</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.40%</td>
      <td>85.39%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>88702</td>
      <td>88699</td>
      <td>-3</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15168</td>
      <td>15171</td>
      <td>+3</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24589/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24589/coverage/framework)


## Modules coverage

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>968eb4</th>
      <th colspan=4>#24589 a01936</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.38%</td>
      <td>90.39%</td>
      <td>+0.00%</td>
      <td>96.26%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>12544</td>
      <td>12551</td>
      <td>+7</td>
      <td>309</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1335</td>
      <td>1335</td>
      <td>-</td>
      <td>12</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24589/coverage_diff/thermal_hydraulics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24589/coverage/thermal_hydraulics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24589/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud committed:
Address CIVET failures:
-doco folder and src folder must match
-duplicate a FP page BISON referencing that page to get that doc to build

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/controllogic/PIDControl.md:
```diff
@@ -0,0 +1,22 @@
+
+!equation
+\text{error} = \text{set point} - \text{value} \\
+\text{output} = K_p * \text{error} + \text{error integral} + K_d * \dfrac{\text{error} - \text{error}_{old}}{dt};
```
Need to mention K_i as well.

GiudGiud replied:
done

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/controllogic/CopyPostprocessorValueControl.md:
```diff
@@ -0,0 +1,12 @@
+
+!syntax description /Controls/CopyPostprocessorValueControl
+
+This is useful for the numerous `ControlLogic` objects that rely on control data. Using this
```
```suggestion
This is useful for the numerous [ControlLogic](ControlLogic/index.md) objects that rely on control data. Using this
```

joshuahansel commented:
Note that I can help with addressing some of my comments, but it wouldn't be for another week or so.

GiudGiud committed:
Address Josh's review

Add doc page for Control data
Link to controllogic more

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

GiudGiud commented on modules/thermal_hydraulics/doc/content/source/controllogic/TerminateControl.md:
```diff
@@ -0,0 +1,25 @@
+
+!syntax description /Controls/TerminateControl
+
+The boolean that determines when to terminate is a `ControlData`. This is only used in the
```
done.
I've added a bunch of links, see latest commit

GiudGiud commented:
BISON doco patch is created and approved.

moosebuild commented:
Job [Documentation internal apps](https://civet.inl.gov/job/1572303/) on eb2fd8e : invalidated by @GiudGiud

patch merged
